src/connection/server_support.c

```diff
@@ -1363,7 +1365,14 @@ css_init (THREAD_ENTRY * thread_p, char *server_name, int name_length, int port_
 
   css_Server_connection_socket = INVALID_SOCKET;
 
-  conn = css_connect_to_master_server (port_id, server_name, name_length);
+  /*
+   * The packing of the server name and type is done by setting the first
+   * character to the numer related to the server type enum value, the rest of the
+   * buffer space is used to copy the server name.
+   */
+  buffer = (char) (get_server_type () + '0');
+  buffer += server_name;
```

Comment:
Append all name_length characters. sometimes several strings are concatenated at the end of server_name, each separated by terminator character.
```suggestion
  buffer.append (server_name, name_length);
```